java/configuration/src/main/java/sleeper/configuration/properties/table/TablePropertyGroup.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright 2022-2023 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package sleeper.configuration.properties.table;
+
+import sleeper.configuration.properties.PropertyGroup;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+import static sleeper.configuration.properties.PropertyGroup.group;
+
+public class TablePropertyGroup {
+    private TablePropertyGroup() {
+    }
+
+    private static final List<PropertyGroup> ALL = new ArrayList<>();
+    public static final PropertyGroup CONFIGURATION = tableGroup("Configuration")
+            .description("The following properties relate to configuring tables.").build();
```

Comment:
This reads like it could include any table property. Maybe split it up a bit more? I think all the properties in this group would fit into either "Data Storage" or "Data Definition".